src/components/tag/tag.jsx

```diff
@@ -0,0 +1,97 @@
+import PropTypes from 'prop-types';
+import React, { useEffect, useRef, useState } from 'react';
+import styles from './tag.css';
+import clearIcon from '../../svgs/icon-clear-small.svg';
+import Icon from '../icon/index.js';
+
+export default function Tag(props) {
+  const [tabActiveStates, setTabActiveStates] = useState([true, false]);
+  const [inputHandled, setInputHandled] = useState(true);
+  const iconElement = useRef(null);
+  const titleElement = useRef(null);
+  const refElements = [titleElement, iconElement];
+
+  function handleGridCellKeyDown(keyEvent) {
```

Comment:
It looks like we are handling active states--that is, focus--according to an array here. What if we switched that piece of state over to a string instead, and then had a helper function indicate whether we were here or there? Something like

```jsx
const [currentTab, setCurrentTab] = useState('tag')
// ...
const isIconActive = () => currentTab === 'icon';
// ...

// Then instead of tabActiveStates[1] we have
if (isIconActive()) {}
```

We could then extract that all to a separate hook and configure it for other tag-like components.

I won't demand such a change--this seems to work alright--I suppose it just smells a bit [primitive obsessive](https://refactoring.guru/smells/primitive-obsession) to me. As always though, take my thoughts with a grain of salt.

Reply:
That's good food for thought! I know we're going to be revisiting this pattern when we implement the list in a dropdown (since list items can also be keyboard navigable) -- and again when we want to DRY up accessibility functionality. 

For future us: I do want to revisit the data structure. I think we're going to want a dynamic array but perhaps not an array of booleans but maybe an array of refs? It expresses better intent (like using strings for static list of items).

Reply:
I'm not sure this could really be called primitive obsession though, because the only state we need to represent (for now) is the `focused` state, which is a boolean value.

Reply:
That's true--but it's for the focused state across two elements, and one has to mentally remember that the first thing is the text and the second thing is the icon. Perhaps instead of a string an object with keys as labels for what is/isn't focused may be a nice refactor:

```jsx
const [currentTab, setCurrentTab] = useState({ tag: false, icon: false });
```

Or better yet, an object that has a string as its interface:

```jsx
const [currentTab, setCurrentTab] = useState('tag');
// currentTab = { tag: false, icon: false }
```